lib/modules/manager/npm/post-update/rules.ts

```diff
@@ -3,51 +3,83 @@ import * as hostRules from '../../../../util/host-rules';
-      uri =
-        is.string(uri) && isHttpUrl(uri)
-          ? uri.replace(regEx(/^https?:/), '')
-          : // TODO: types (#22198)
-            `//${uri}/`;
-      if (hostRule.token) {
-        const key = hostRule.authType === 'Basic' ? '_auth' : '_authToken';
-        additionalNpmrcContent.push(`${uri}:${key}=${hostRule.token}`);
-        additionalYarnRcYml ||= { npmRegistries: {} };
-        if (hostRule.authType === 'Basic') {
-          additionalYarnRcYml.npmRegistries[uri] = {
-            npmAuthIdent: hostRule.token,
-          };
-        } else {
-          additionalYarnRcYml.npmRegistries[uri] = {
-            npmAuthToken: hostRule.token,
-          };
-        }
-      } else if (is.string(hostRule.username) && is.string(hostRule.password)) {
-        const password = toBase64(hostRule.password);
-        additionalNpmrcContent.push(`${uri}:username=${hostRule.username}`);
-        additionalNpmrcContent.push(`${uri}:_password=${password}`);
-        additionalYarnRcYml ||= { npmRegistries: {} };
-        additionalYarnRcYml.npmRegistries[uri] = {
-          npmAuthIdent: `${hostRule.username}:${hostRule.password}`,
+    if (!hostRule.resolvedHost) {
+      continue;
+    }
+
+    const matchedHost = hostRule.matchHost;
+    // Should never be needed as if we have a resolvedHost there has to be a matchHost
+    // istanbul ignore next
+    if (!matchedHost) {
+      continue;
+    }
+
+    const uri = `//${matchedHost}/`;
+    let cleanedUri = uri;
+    if (is.string(matchedHost) && isHttpUrl(matchedHost)) {
+      cleanedUri = matchedHost.replace(regEx(/^https?:/), '');
+    }
+
+    if (hostRule.token) {
+      const key = hostRule.authType === 'Basic' ? '_auth' : '_authToken';
+      additionalNpmrcContent.push(`${cleanedUri}:${key}=${hostRule.token}`);
+
+      if (hostRule.authType === 'Basic') {
+        const registry = {
+          npmAuthIdent: hostRule.token,
         };
+        additionalYarnRcYml.npmRegistries[cleanedUri] = registry;
+        additionalYarnRcYml.npmRegistries[uri] = registry;
+
+        continue;
       }
+
+      const registry = {
+        npmAuthToken: hostRule.token,
+      };
+      additionalYarnRcYml.npmRegistries[cleanedUri] = registry;
+      additionalYarnRcYml.npmRegistries[uri] = registry;
+
+      continue;
+    }
+
+    if (is.string(hostRule.username) && is.string(hostRule.password)) {
+      const password = toBase64(hostRule.password);
+      additionalNpmrcContent.push(
+        `${cleanedUri}:username=${hostRule.username}`,
+      );
+      additionalNpmrcContent.push(`${cleanedUri}:_password=${password}`);
+
+      const registries = {
+        npmAuthIdent: `${hostRule.username}:${hostRule.password}`,
+      };
+      additionalYarnRcYml.npmRegistries[cleanedUri] = registries;
+      additionalYarnRcYml.npmRegistries[uri] = registries;
     }
   }
-  return { additionalNpmrcContent, additionalYarnRcYml };
+
+  const hasYarnRCNPMRegistries =
```

Comment:
```suggestion
  const hasYarnRcNpmRegistries =
```
be consistent with interface